automatic_symbol_loading/automatic_symbol_loading.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+from pathlib import Path
+import gdb
+
+try:
+    from elftools.elf.elffile import ELFFile
+except ModuleNotFoundError:
+    check_build_id = False
+else:
+    check_build_id = True
+
+def create_file_dict(path):
+    ret_dict = {}
+    p = sorted(Path(path).glob('**/*.debug'))
+    for f in p:
+        # Remove the .debug suffix from the file name
+        ret_dict[f.name[:-6]] = f
+    return ret_dict
+
+
+def match_build_id(debug_file, obj):
+    if not check_build_id:
+        return True
+    with debug_file.open('rb') as fd:
+        ef = ELFFile(fd)
+        sect = ef.get_section_by_name('.note.gnu.build-id')
+        # The data is a long string that contains the build-id at the end,
+        # starting from index 32. Reference can be found here:
+        # https://interrupt.memfault.com/blog/gnu-build-id-for-firmware
+        # in hex each byte is 2 char long -> 4fields*4bytes*2char = 32
+        # NOTE: We always assume that name is 4 bytes "GNU\0".
+        # This is always true for GCC and Clang compiled binaries.
+        debug_build_id = sect.data().hex()[32:]
+    return debug_build_id == obj.build_id
+
+
+class ExtraSymbolsCommand(gdb.Command):
+    """
+    A command to load all external symbol files for debuggee.
+    Usage: allsym PATH_TO_SYMBOL_ROOT
+    """
+
+    def __init__(self):
+        super().__init__("load-all-symbols", gdb.COMPLETE_FILENAME)
```

Comment:
It's still wrong. The completer argument is the third argument. The second indicates what kind of command this is.
Looking at https://sourceware.org/gdb/current/onlinedocs/gdb/Commands-In-Python.html, I think it should be a support command (`set sysroot` is a support command for instance) so:
```
super().__init__("load-all-symbols", gdb.COMMAND_SUPPORT, gdb.COMPLETE_FILENAME)
```

Reply:
Apologies, I missed it. I think COMMAND_FILES might be better suited possibly as this is effectively mingling with files... what do you think?